snuba/utils/metrics/backends/dummy.py

```diff
@@ -1,9 +1,37 @@
-from collections import Mapping
-from typing import Optional, Union
+from __future__ import annotations
+
+from typing import List, Mapping, MutableMapping, Optional, Tuple, Union
 
 from snuba.utils.metrics.backends.abstract import MetricsBackend
 from snuba.utils.metrics.types import Tags
 
+RecordedMetricCalls = List[Tuple[Union[int, float], Tags]]
+
+RECORDED_METRIC_CALLS: MutableMapping[
+    str, MutableMapping[str, RecordedMetricCalls]
+] = {}
+
+
+def record_metric_call(
+    mtype: str, name: str, value: int | float, tags: Optional[Tags]
+) -> None:
+    if mtype not in RECORDED_METRIC_CALLS:
+        RECORDED_METRIC_CALLS[mtype] = {}
+
+    if name not in RECORDED_METRIC_CALLS[mtype]:
+        RECORDED_METRIC_CALLS[mtype][name] = []
+
+    if tags is None:
+        tags = {}
+    RECORDED_METRIC_CALLS[mtype][name].append((value, tags))
+
+
+def get_recorded_metric_calls(mtype: str, name: str) -> RecordedMetricCalls | None:
```

Comment:
Having this be one global variable across all tests will make it so that different tests pollute each others' results. I have two different solutions to this problem:

1. Clear the global dictionary in the conftest.py before every test
2. Instead of keeping track of this state ourselves we:

1. Make the DummyMetricsBackend log each metric under a its own logger 
```python3
metrics_logger = logging.get_logger("dummy.metrics")
```
2. Log each metric as a dictionary:
```python3
metrics_logger.info({"name": ...})
```
3. use the caplog fixture in pytest to get the logs specific to a test

Reply:
I decided to reset it for now. If this starts to become a pervasive testing strategy we can simplify it.